api/controllers/user.go

```diff
@@ -0,0 +1,94 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+	"net/mail"
+
+	"github.com/PedroChaparro/loomies-backend/interfaces"
+	"github.com/PedroChaparro/loomies-backend/models"
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/mongo"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func HandleSignUp(c *gin.Context) {
+	var err error
+	var form interfaces.SignUpForm
+
+	if err := c.BindJSON(&form); err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Bad request"})
+		return
+	}
+
+	err = models.CheckExistEmail(form.Email)
+
+	if !(err != nil) {
+		if !(err == mongo.ErrNoDocuments) {
+			c.AbortWithStatusJSON(http.StatusConflict,
+				gin.H{"message": "Email already exists"})
+			return
+		}
+	}
+
+	err = models.CheckExistUser(form.User)
+
+	if !(err != nil) {
+		if !(err == mongo.ErrNoDocuments) {
+			c.AbortWithStatusJSON(http.StatusConflict,
+				gin.H{"message": "User already exists"})
+			return
+		}
+	}
+
+	if form.User == "" {
```

Comment:
Empty values (username, mail and password) should be validated before querying the database to avoid unnecessary queries.

Reply:
Wouldn't it be better to validate on both the client and the server?